Sources/Logging/Logging.swift

```diff
@@ -251,34 +251,57 @@ extension Logger {
     }
 }
 
+/// The `LogHandlerFactory` is the bridge between the `LoggingSystem` and the metrics backend implementation.
+/// `LogHandlerFactory` role is to initialize concrete `LogHandler` instances,
+/// it may also contain any additional state it wishes to keep for operating those handlers if it needs to.
+///
+/// This type is an implementation detail and should not be used directly, unless implementing your own metrics backend.
+/// To use the Logging API, please refer to the documentation of `LoggingSystem`.
+public protocol LogHandlerFactory {
+    func make(label: String) -> LogHandler
+}
+
+//// Note: Other implementations can also offer their factories like this as an extension for easier discoverability.
+//public extension LogHandlerFactory {
+//    /// Convenience for creating simple ad-hoc log handler factories which are easily expressed as simple closures.
+//    static func makeFromClosure(_ closure: @escaping (String) -> LogHandler) -> LogHandlerFactory {
+//        return ClosureLogHandlerFactory(closure)
+//    }
```

Comment:
TODO: Remove this, was playing around if we could provide some convenience but sadly can't put it like that on a protocol... Did I miss any good ideas™ here?